packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import List from 'terra-list';
+import ActionHeader from 'terra-clinical-action-header';
+import ContentContainer from 'terra-content-container';
+import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
+
+import styles from './NavigationSideMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+};
+
+const propTypes = {
+  /**
+   * An array of configuration for each menu item.
+   */
+  menuItems: PropTypes.arrayOf(PropTypes.shape({
+    /**
+     * Keys of menu items
```

Comment:
Are the children really the keys?

Reply:
Menu items are a flat list of all possible menus. We parse the array and create an object, in order to have quick access to the individual item.